internal/pkg/service/buffer/storage/local/writer/base/base.go

```diff
@@ -1,31 +1,49 @@
 package base
 
 import (
+	"bufio"
+	"github.com/benbjohnson/clock"
+	"github.com/c2h5oh/datasize"
 	"github.com/keboola/keboola-as-code/internal/pkg/log"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/compression"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local/writer/disksync"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local/writer/writechain"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model/column"
+	"io"
 )
 
+const fileBufferSize = 64 * datasize.KB
+
+type chain = writechain.Chain
+
+type syncer = disksync.Syncer
+
 type Writer struct {
+	*chain
+	*syncer
 	logger   log.Logger
 	slice    *storage.Slice
 	dirPath  string
 	filePath string
-	chain    *writechain.Chain
-	syncer   *disksync.Syncer
 }
 
-func NewWriter(logger log.Logger, slice *storage.Slice, dirPath string, filePath string, chain *writechain.Chain, syncer *disksync.Syncer) Writer {
-	return Writer{
+func NewWriter(logger log.Logger, clock clock.Clock, slice *storage.Slice, dirPath string, filePath string, chain *writechain.Chain) *Writer {
+	w := &Writer{
+		chain:    chain,
+		syncer:   disksync.NewSyncer(logger, clock, slice.LocalStorage.Sync, chain),
 		logger:   logger,
 		slice:    slice,
 		dirPath:  dirPath,
 		filePath: filePath,
-		chain:    chain,
-		syncer:   syncer,
 	}
+
+	// Add a small buffer before the file
+	w.chain.PrependWriter(func(writer writechain.Writer) io.Writer {
+		return bufio.NewWriterSize(writer, int(fileBufferSize.Bytes()))
+	})
```

Comment:
Pribudol tu maly `64KB` bufffer (pred cielovy *os.File).
- Experimentalne zistene v benchmarkoch, ze je to asi o `10%` rýchlejsie.
- Zápis do súboru je `syscall`, a je efektívnejšie ho volať menej často s väčším počtom dát.
- Zapisuje sa najprv do `disk cache` v memory, pri `Sync` (alebo automaticky neskor) priamo na disk.